controllers/storagecluster/job_templates_test.go

```diff
@@ -0,0 +1,135 @@
+package storagecluster
+
+import (
+	"context"
+	"testing"
+
+	openshiftv1 "github.com/openshift/api/template/v1"
+	api "github.com/openshift/ocs-operator/api/v1"
+	"github.com/stretchr/testify/assert"
+	batchv1 "k8s.io/api/batch/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+var expectedOcsRemovalParameterName = "FAILED_OSD_IDS"
+var expectedOcsExtendParameterName = "RECONFIGURE"
+
+var extendClusterCommandTest = []string{
+	"bash",
+	"-c",
+	`full_ratio_value=$(ceph osd dump | grep -m1 full_ratio | awk '{print $2}')
+	  echo $full_ratio_value
+	if [[ $full_ratio_value == "0.85"  ]]; then
+		   ceph osd set-full-ratio 0.87
+	else
+		   ceph osd set-full-ratio 0.85
+	fi
+	`,
+}
+var osdCleanupArgsTest = []string{
+	"ceph",
+	"osd",
+	"remove",
+	"--osd-ids=${FAILED_OSD_IDS}",
+}
+
+func TestTemplateCreated(t *testing.T) {
+
+	t, reconciler, cr, request := initStorageClusterResourceCreateUpdateTestWithPlatform(
+		t, nil, nil)
+	assertOcsRemovalTemplate(t, reconciler, cr, request)
+	assertOcsClusterExtendTemplate(t, reconciler, cr, request)
+	assertOcsClusterExtendJob(t, reconciler, cr, request)
+	assertOcsRemovalJob(t, reconciler, cr, request)
+	expectedTemplates(t, reconciler, cr, request)
+}
+
+func assertOcsRemovalTemplate(t *testing.T, reconciler StorageClusterReconciler, cr *api.StorageCluster, request reconcile.Request) {
+
+	ocsRemovalTemplate := &openshiftv1.Template{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "ocs-osd-removal",
+			Namespace: cr.Namespace,
+		},
+	}
+	request.Name = "ocs-osd-removal"
+	reconciler.Client.Get(context.TODO(), request.NamespacedName, ocsRemovalTemplate)
+	assert.Equal(t, ocsRemovalTemplate.Parameters[0].Name, expectedOcsRemovalParameterName)
```

Comment:
A little clunky, it's usually a good idea to avoid using static index references into arrays for testing. But this works for now.